src/db/add_package.rs

```diff
@@ -328,31 +327,103 @@ fn add_authors_into_database(
     Ok(())
 }
 
+pub fn update_crate_data_in_database(
+    conn: &Connection,
+    name: &str,
+    registry_data: &CrateData,
+) -> Result<()> {
+    info!("Updating crate data for {}", name);
+    let crate_id = conn
+        .query("SELECT id FROM crates WHERE crates.name = $1", &[&name])?
+        .get(0)
+        .get(0);
+
+    update_owners_in_database(conn, &registry_data.owners, crate_id)?;
+
+    Ok(())
+}
+
 /// Adds owners into database
-fn add_owners_into_database(conn: &Connection, owners: &[CrateOwner], crate_id: i32) -> Result<()> {
+fn update_owners_in_database(
+    conn: &Connection,
+    owners: &[CrateOwner],
+    crate_id: i32,
+) -> Result<()> {
+    let existing_owners: Vec<String> = conn
+        .query(
+            "
+        SELECT login
+        FROM owners
+        INNER JOIN owner_rels
+            ON owner_rels.oid = owners.id
+        WHERE owner_rels.cid = $1
+    ",
+            &[&crate_id],
+        )?
+        .into_iter()
+        .map(|row| row.get(0))
+        .collect();
```

Comment:
You only use this later as an iterator, so the `collect()` isn't necessary.
```suggestion
        .map(|row| row.get(0));
```